pkg/kubelet/dockertools/manager.go

```diff
@@ -288,8 +288,15 @@ func (dm *DockerManager) inspectContainer(dockerID, containerName, tPath string)
 		result.status.State.Running = &api.ContainerStateRunning{
 			StartedAt: util.NewTime(inspectResult.State.StartedAt),
 		}
-		if containerName == PodInfraContainerName && inspectResult.NetworkSettings != nil {
-			result.ip = inspectResult.NetworkSettings.IPAddress
+		if dm.networkPlugin.Name() != network.DefaultPluginName && containerName == PodInfraContainerName {
```

Comment:
I like this proposed layout better but it's a minor point - if no other changes are necessary I won't require it :)